vision/src/autogluon/vision/predictor/predictor.py

```diff
@@ -469,17 +498,25 @@ def predict(self, data, as_pandas=True):
         """
         if self._classifier is None:
             raise RuntimeError('Classifier is not initialized, try `fit` first.')
+        assert self._label_cleaner is not None
         proba = self._classifier.predict(data)
         if 'image' in proba.columns:
             # multiple images
-            ret = proba.loc[proba.groupby(["image"])["score"].idxmax()].reset_index(drop=True)
+            assert isinstance(data, pd.DataFrame) and 'image' in data.columns
+            y_pred = proba.loc[proba.groupby(["image"])["score"].idxmax()].reset_index(drop=True)
```

Comment:
Don't fix in this PR, but long-term we don't want to be doing this kind of computation, as it is likely a significant increase to inference latency. I suspect changing gluoncv to output the ideal format in the first place might make things a lot simpler.